frontend/src/components/common/ImpactPerCause.tsx

```diff
@@ -0,0 +1,149 @@
+import React, { useState, useEffect } from "react";
+import { Box, Text, Flex, Button, Center, Circle } from "@chakra-ui/react";
+
+
+interface ItemData {
+  item_id: number;
+  item_name: string;
+  total_impact: number;
+}
+
+interface CauseData {
+  cause_id: number;
+  cause_name: string;
+  items: ItemData[]; 
+}
+
+const ImpactPerCause: React.FC = () => {
+  const [pageIndex, setPageIndex] = useState(0);
+  const [causes, setCauses] = useState<CauseData[]>([]);
+
+  // Buttons to move between different causes/pages
+  const handlePrevious = () => {
+    setPageIndex((prev) => (prev > 0 ? prev - 1 : prev));
+  };
+
+  const handleNext = () => {
+    setPageIndex((prev) => (prev < causes.length - 1 ? prev + 1 : prev));
+  };
+
+  // As of right now, impact is fetched from this link. In the future, this will be replaced by authContext.
+  // Change to localhost:5000 and modify the userId below based on the data entered in Prisma
+  async function fetchImpactData(userId: string) {
+    try {
+      const response = await fetch(`http://localhost:5001/impacts/${userId}`);
```

Comment:
Can you pass in the frontend url from the env file instead of hardcoding it here?

That way, when we change the frontend url in the env file, this dashboard page will call the updated frontend url.

Reply:
I think you guys hardcoded the frontend url in a few other places too — if you could dynamically pass them in from the env file too 🙏